hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineProvider.java

```diff
@@ -77,8 +82,14 @@ public RatisPipelineProvider(NodeManager nodeManager,
     this.conf = conf;
     this.eventPublisher = eventPublisher;
     this.scmContext = scmContext;
-    this.placementPolicy =
-        new PipelinePlacementPolicy(nodeManager, stateManager, conf);
+    try {
+      this.placementPolicy = PipelinePlacementPolicyFactory
+          .getPolicy(conf, nodeManager, stateManager,
+              nodeManager.getClusterNetworkTopologyMap(), true, null);
+    } catch (Exception e) {
+      this.placementPolicy = null;
+      LOG.info("Cannot create pipeline policy for pipeline, {}", e);
```

Comment:
Just throw the exception out.  Otherwise, it will generate NPE later.

Reply:
Fixed.